internal/sapsystem/sapcontrol/webservice.go

```diff
@@ -0,0 +1,150 @@
+package sapcontrol
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net"
+	"net/http"
+	"path"
+
+	"github.com/hooklift/gowsdl/soap"
+)
+
+//go:generate mockery --all
+
+type WebService interface {
+	GetInstanceProperties() (*GetInstancePropertiesResponse, error)
+	GetProcessList() (*GetProcessListResponse, error)
+	GetSystemInstanceList() (*GetSystemInstanceListResponse, error)
+}
+
+type STATECOLOR string
+type STATECOLOR_CODE int
+
+const (
+	STATECOLOR_GRAY        STATECOLOR      = "SAPControl-GRAY"
+	STATECOLOR_GREEN       STATECOLOR      = "SAPControl-GREEN"
+	STATECOLOR_YELLOW      STATECOLOR      = "SAPControl-YELLOW"
+	STATECOLOR_RED         STATECOLOR      = "SAPControl-RED"
+	STATECOLOR_CODE_GRAY   STATECOLOR_CODE = 1
```

Comment:
I think we don't use these `int` values by now. Maybe we should still keep them, but we could add a comment on top of them saying that they were just carried over form the `sap_host_exporter` project